src/Compilers/Core/Portable/InternalUtilities/ReadOnlyMemoryOfCharComparer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class ReadOnlyMemoryOfCharComparer : IEqualityComparer<ReadOnlyMemory<char>>
+    {
+        public static ReadOnlyMemoryOfCharComparer Instance { get; } = new();
+
+        private ReadOnlyMemoryOfCharComparer()
+        {
+        }
+
+        public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
+            => x.Span.SequenceEqual(y.Span);
+
+        public int GetHashCode(ReadOnlyMemory<char> obj)
+        {
+#if NET
```

Comment:
I think we should get rid of the conditional compilation here and simply use ```Hash.GetFNVHashCode```. If we feel that ```string.GetHashCode``` is definitely superior, then we should consider adjusting implementation of ```Hash.GetFNVHashCode``` accordingly, but do that separately (not in context of this PR).

Reply:
so the string versions are scrutinized and optimized.  They both optimize for small strings, and they batch characters together.  So i'd prefer to not leave perf on the table wrt .net core.